### PR TITLE
fix(rust, python): raise if tz_localize called on UTC-aware

### LINKED ISSUE #6525: tz_localize is allowed (but ignored?) when current timezone is UTC?
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Issue description

Calling `tz_localize` on a tz-aware Series errors. Unless it's UTC, in which case the call is just ignored?

### Reproducible example

```
In [2]: ts_utc = pl.Series(['2018-10-28']).str.strptime(pl.Datetime).dt.tz_localize('UTC')

In [3]: ts_utc
Out[3]:
shape: (1,)
Series: '' [datetime[μs, UTC]]
[
        2018-10-28 00:00:00 UTC
]

In [4]: ts_utc.dt.tz_localize('US/Pacific')
Out[4]:
shape: (1,)
Series: '' [datetime[μs, UTC]]
[
        2018-10-28 00:00:00 UTC
]
```

### Expected behavior

I think I'd expect
```
ComputeError: Cannot localize a tz-aware datetime. Consider using 'dt.with_time_zone' or 'dt.cast_time_zone'
```

Either that, or to set the new timezone. Probably the former (error). 

### Installed versions

<details>
---Version info---
Polars: 0.15.18
Index type: UInt32
Platform: Linux-5.10.102.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python: 3.11.1 (main, Dec  7 2022, 01:11:34) [GCC 11.3.0]
---Optional dependencies---
pyarrow: <not installed>
pandas: 1.5.3
numpy: 1.24.1
fsspec: <not installed>
connectorx: <not installed>
xlsx2csv: <not installed>
deltalake: <not installed>
matplotlib: <not installed>

</details>

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #6525

### CODE REVIEW

File: polars/polars-lazy/polars-plan/src/dsl/function_expr/datetime.rs [hunk 1]
```diff
@@ -146,10 +146,7 @@ pub(super) fn cast_timezone(s: &Series, tz: &str) -> PolarsResult<Series> {
 pub(super) fn tz_localize(s: &Series, tz: &str) -> PolarsResult<Series> {
     let ca = s.datetime()?.clone();
     match ca.time_zone() {
-        Some(tz) if tz == "UTC" => {
-           Ok(ca.with_time_zone(Some("UTC".into()))?.into_series())
-        }
```
Comment (MarcoGorelli) [inline polars/polars-lazy/polars-plan/src/dsl/function_expr/datetime.rs:151]: This was added in https://github.com/pola-rs/polars/pull/5522/files , but it looks like it's no longer necessary

Comment (MarcoGorelli) [reply polars/polars-lazy/polars-plan/src/dsl/function_expr/datetime.rs:151]: Hmm maybe the intention was to match the target tz? I'll update later, marking as draft for now

Comment (MarcoGorelli) [reply polars/polars-lazy/polars-plan/src/dsl/function_expr/datetime.rs:151]: I've rewritten according to what I think the intention was

**Comment by ritchie46** [comment]
Yeap.. :+1:

**Pull request merged**